### PR TITLE
listen to editor events and set active and dirty flags accordingly

### PR DESCRIPTION
**Pull request by franzbecker**

### CODE REVIEW

File: src/lib/src/component/tree-viewer/tree-viewer.component.css [hunk 1]
```diff
@@ -32,4 +32,16 @@
 
 .tree-view-item-key:hover {
     background-color: #2A2D2E;
+}
+
+.active {
+    background-color: #3f3f46 !important;
```
Comment (wiedehoeft) [inline src/lib/src/component/tree-viewer/tree-viewer.component.css]: Would check if this is really needed, its a hack, http://www.css-hack.de/CSS+allgemein/!+important:

"Bei großen Projekten oder für die schnelle Lösung kommen Sie manchmal nicht ohne ein ! important aus. Eleganter ist es aber immer, eine neue Klasse einzuführen, oder eine saubere Cascade zu schreiben."

File: src/lib/src/component/tree-viewer/tree-viewer.component.events.spec.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import { WorkspaceElement } from '../../service/persistence/workspace-element';
+import * as events from './event-types';
+import { testBedSetup } from './tree-viewer.component.spec';
+
+describe('TreeViewerComponent event handling', () => {
+
+  let component: TreeViewerComponent;
+  let fixture: ComponentFixture<TreeViewerComponent>;
+  let messagingService: MessagingService;
+
```
Comment (wiedehoeft) [inline src/lib/src/component/tree-viewer/tree-viewer.component.events.spec.ts:17]: Do you test with every browser? It seems that the "let" support is not as good as you would expect to be:
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Statements/let

File: src/lib/src/component/tree-viewer/tree-viewer.component.css (hunk 1 above)
Comment (franzbecker) [reply src/lib/src/component/tree-viewer/tree-viewer.component.css]: Yes, thanks we can exclude it with `.tree-view-item-key:not(.active):hover` for the hover styling.

File: src/lib/src/component/tree-viewer/tree-viewer.component.events.spec.ts (hunk 2 above)
Comment (franzbecker) [reply src/lib/src/component/tree-viewer/tree-viewer.component.events.spec.ts:17]: That's a valid concern but luckily TypeScript got us covered. We compile to ES5 here so the following code in TypeScript:
```
let isOwnElement = element.path === this.model.path;
```
compiles to:
```
var isOwnElement = element.path === _this.model.path;
```

**Review by wiedehoeft — APPROVED** [review]

**Pull request merged**